src/services/tier/useTier.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+
+export const TierNames = {
+  BASIC: 'basic',
+  LITE: 'lite',
+  PRO: 'pro',
+  ENTERPRISE: 'enterprise',
+} as const
+
+export const TierSchema = z
+  .object({
+    tierName: z.nativeEnum(TierNames),
+  })
+  .nullish()
```

Comment:
Can you use this as a child of a `RequestSchema` and parse all the way down from the owner to make sure we're validating the whole response.